tests/chart_tests/test_vector.py

```diff
@@ -0,0 +1,294 @@
+        all namespaces."""
+        values = {
+            "global": {
+                "logging": {"collector": "vector"},
+                "manualNamespaceNamesEnabled": True,
+                "features": {
+                    "namespacePools": {
+                        "enabled": False,
+                    }
+                },
+            }
+        }
+
+        doc = render_chart(
+            kube_version=kube_version,
+            values=values,
+            show_only=["charts/vector/templates/vector-configmap.yaml"],
+        )[0]
+
+        expected_rule = "match(.kubernetes.pod_namespace, r'^[a-z0-9]([-a-z0-9]*[a-z0-9])?$')"
+        assert expected_rule in doc["data"]["vector-config.yaml"]
+
+    def test_vector_configmap_manual_namespaces_and_namespacepools_disabled(self, kube_version):
+        """Test that when namespace Pools and manualNamespaceNamesEnabled are disabled, helm renders a default vector configmap
+        looking at platform label."""
+        values = {
+            "global": {
+                "manualNamespaceNamesEnabled": False,
+                "logging": {"collector": "vector"},
+                "features": {
+                    "namespacePools": {
+                        "enabled": False,
+                    }
+                },
+            }
+        }
+        doc = render_chart(
+            kube_version=kube_version,
+            values=values,
+            show_only=["charts/vector/templates/vector-configmap.yaml"],
+        )[0]
+        expected_rule = '.kubernetes.namespace_labels.platform == "release-name"'
+        assert expected_rule in doc["data"]["vector-config.yaml"]
+
+    def test_vector_securityContext_override(self, kube_version):
+        """Test that helm renders a custom securityContext when securityContext is overridden."""
+
+        values = {
+            "global": {"logging": {"collector": "vector"}},
+            "vector": {
+                "podSecurityContext": {"happy": "family"},  # pod securityContext
+                "vector": {"securityContext": {"runAsUser": 9999}},  # container securityContext
+            },
+        }
+
+        docs = render_chart(
+            kube_version=kube_version,
+            values=values,
+            show_only=["charts/vector/templates/vector-daemonset.yaml"],
+        )
+        assert len(docs) == 1
+
+        pod_spec = docs[0]["spec"]["template"]["spec"]
+        assert pod_spec["securityContext"] == {"happy": "family"}
+        assert len(pod_spec["containers"]) == 1
+        assert pod_spec["containers"][0]["securityContext"]["runAsUser"] == 9999
+
+    def test_vector_securityContext_default(self, kube_version):
+        """Test that we have the expected default security context for the vector container."""
+
+        values = {"global": {"logging": {"collector": "vector"}}}
+
+        docs = render_chart(
+            kube_version=kube_version,
+            values=values,
```

Comment:
If the above is "yes" (which i'm 99% sure it is) then we need to delete this too.